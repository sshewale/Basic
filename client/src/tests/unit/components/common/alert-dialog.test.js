import React from 'react';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';
import { Dialog, DialogContent, Button } from '@material-ui/core';
import AlertDialog from '../../../../components/common/alert-dialog/alert-dialog';

configure({ adapter: new Adapter() });

const dialogProps = {
  onClose: jest.fn(),
  onClick: jest.fn(),
  alertMessage: 'No records found to export.',
  openDialog: true,
};

describe('alert dialog component', () => {
  let alertDialog;
  beforeEach(() => {
    alertDialog = shallow(<AlertDialog />);
    alertDialog.setProps(dialogProps);
  });

  it('should not open alert without the required props', () => {
    alertDialog.setProps({});
    expect.assertions(1);
    expect(toJson(alertDialog)).toMatchSnapshot();
  });

  it('should render without crashing given the required props', () => {
    expect.assertions(1);
    expect(toJson(alertDialog)).toMatchSnapshot();
  });

  it('should render diaglog component', () => {
    expect.assertions(2);
    const dialogComponent = alertDialog.find(Dialog);
    expect(dialogComponent.length).toEqual(1);
    expect(dialogComponent.props().open).toBe(true);
  });

  it('dialog should render DialogContent component with text valie as "No records found to export."', () => {
    expect.assertions(2);
    const dialogContentComponent = alertDialog.find(Dialog).find(DialogContent);
    expect(dialogContentComponent.length).toEqual(1);
    expect(dialogContentComponent.text()).toEqual(
      'No records found to export.',
    );
  });

  it('dialog should render Button component with text value as "Ok"', () => {
    const buttonComponent = alertDialog.find(Dialog).find(Button);
    expect(buttonComponent.length).toEqual(1);
    expect(buttonComponent.text()).toEqual('Ok');
  });
});
