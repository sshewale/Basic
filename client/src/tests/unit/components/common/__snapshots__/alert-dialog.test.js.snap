// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`alert dialog component should not open alert without the required props 1`] = `
<WithStyles(ForwardRef(Dialog))
  onClose={[MockFunction]}
  open={true}
>
  <WithStyles(ForwardRef(DialogContent))>
    <WithStyles(ForwardRef(DialogContentText))>
      No records found to export.
    </WithStyles(ForwardRef(DialogContentText))>
  </WithStyles(ForwardRef(DialogContent))>
  <WithStyles(ForwardRef(Button))
    color="primary"
    onClick={[MockFunction]}
  >
    Ok
  </WithStyles(ForwardRef(Button))>
</WithStyles(ForwardRef(Dialog))>
`;

exports[`alert dialog component should render without crashing given the required props 1`] = `
<WithStyles(ForwardRef(Dialog))
  onClose={[MockFunction]}
  open={true}
>
  <WithStyles(ForwardRef(DialogContent))>
    <WithStyles(ForwardRef(DialogContentText))>
      No records found to export.
    </WithStyles(ForwardRef(DialogContentText))>
  </WithStyles(ForwardRef(DialogContent))>
  <WithStyles(ForwardRef(Button))
    color="primary"
    onClick={[MockFunction]}
  >
    Ok
  </WithStyles(ForwardRef(Button))>
</WithStyles(ForwardRef(Dialog))>
`;
